File: ED_RRiveraG1_T02/Node.java
package ED.Riverto;

public class Node <T> {
    //Atributos
    Node next; //Apuntador a siguiente nodo.
    T data; //Dato genérico.

    //Declaración de constructores.
    public Node(){
        next= null;
    }
    public Node(T d){
        next=null;
        data = d;
    }

    //Declaración Método mostrar
    public void print(){
        System.out.println(data + "->");
    }
    public void printS(){
        System.out.println(data + "\n ↓");
    }
}

File: ED_Juntos_C/Node.cpp
//
// Created by Roberto Rivera on 10/9/17.
//

#import "iostream"
template <class T> class Node {
public:
    //Atributos
    Node *next; //Apuntador a siguiente nodo.
    T data; //Dato genérico.


    Node() {
        next = nullptr;
    }

//Consturctor con un elemento
    Node(T d) {
        data = d;
        next = nullptr;
    }

//Imprimir Nodo
    void print() {
        std::cout << data.toString() << "->";
    }

//Imprimir Nodo para pila
    void printS() {
        std::cout << data << std::endl << "↓" << std::endl;
    }

//Destructor
    ~Node() {}
};